Unify tests extras_require

idgserpro opened the pull request:
It makes no sense to have two if only one is used in tests.

tdesvenain reviewed (changes requested):
This configuration is necessary to launch tests with archetypes

idgserpro commented:
@tdesvenain see that I kept the `Products.ATContentTypes`:

https://github.com/collective/collective.elasticsearch/blob/bc4c09fb5f18847409a1e2443c865c743c2e7123/setup.py#L62

idgserpro commented:
@tdesvenain can you please check the comment above?

tdesvenain commented:
So you can't test without at content types, can you ?

idgserpro commented:
Yes. It would not be possible to test without archetypes. But today it is already being tested without archetypes:

https://github.com/collective/collective.elasticsearch/blob/c60398a62d7fbd67e2571bf809797dedff75e8f8/test_plone51.cfg#L8

It is necessary to reformulate the tests. I will close here. Thanks.

idgserpro commented:
In fact, tests done without AT:

https://github.com/collective/collective.elasticsearch/blob/9f5d42e031997069ae414dd4b12edab93e0ccaaa/src/collective/elasticsearch/testing.py#L33

and with AT:

https://github.com/collective/collective.elasticsearch/blob/9f5d42e031997069ae414dd4b12edab93e0ccaaa/src/collective/elasticsearch/testing.py#L82

See:

https://github.com/collective/collective.elasticsearch/blob/ff898bab6aca2d17ca7dacd3ec5b91df5111dbb1/src/collective/elasticsearch/tests/test_catalog.py#L84-L88

So, can you please approve the PR?

idgserpro commented:
@tdesvenain the build is green again. Can you take a look here please?

idgserpro commented:
@tdesvenain  I moved the `Products.ATContentTypes` from the extra test-archetypes to extra test. See:

https://github.com/collective/collective.elasticsearch/blob/08e8d369b78d65c779059fcc5ea3bad69bbcbe97/setup.py#L58-L65

Although I removed the `package-extras = [test-archetypes]` from buildout, the extra test, which owned `Products.ATContentTypes` is still loaded in the test.

The part test uses the eggs from instance part:

https://github.com/collective/collective.elasticsearch/blob/8f58491aa9adabf9502c6f6a1805b8a64df4e1f5/base.cfg#L47-L52

instance part eggs include the extra test:

https://github.com/collective/collective.elasticsearch/blob/8f58491aa9adabf9502c6f6a1805b8a64df4e1f5/base.cfg#L23-L32

So, `Products.ATContentTypes` is still used in the tests.

Can you please review and approve this PR? Thanks!
